Add include_geojson and zoom_level to the params of get_geo_centre()

cn1036 opened the pull request:
Status: tested locally, ready for review.

API users will be able to specify whether shapefile needs to be included in the response.
API endpoint changes: https://grointelligence.atlassian.net/browse/CLEWS-27183

cn1036 committed:
add include_geojson and zoom_level to get_geo_centre()

nemozen reviewed (changes requested):
we need zoom level support on get_geojson() not get_geo_centre()

nemozen committed:
zoom_level arg on get_geojsons rather than get_geo_centre

cn1036 commented:
@nemozen 
sorry, working on stage RC right now. will review the commit and update unit tests soon

cn1036 committed:
add param to get_geojson(); update comments

cn1036 commented on api/client/gro_client.py:
```diff
@@ -301,38 +301,48 @@ def get_geo_centre(self, region_id):
         return lib.get_geo_centre(self.access_token, self.api_host, region_id)
 
-    def get_geojsons(self, region_id, descendant_level=None):
+    def get_geojsons(self, region_id, descendant_level=None, zoom_level=7):
```
just to confirm, we used to return simplified shapefile for all requests. is it intended to change the default to detailed boundaries table? @nemozen

nemozen replied:
Yes, we want to keep the current server-side default, so we don't change
client side performance of the web app. for api clients, we want to default
to more detailed boundaries (as with shp files), and give  the client the
option to reduce the zoom level if higher level of accuracy is not needed.

nemozen commented:
Yes, we want to keep the current server-side default, so we don't change
client side performance of the web app. for api clients, we want to default
to more detailed boundaries (as with shp files), and give  the client the
option to reduce the zoom level if higher level of accuracy is not needed.

On Mon, Jun 15, 2020 at 2:51 PM cn1036 <notifications@github.com> wrote:

> *@cn1036* commented on this pull request.
> ------------------------------
>
> In api/client/gro_client.py
> <https://github.com/gro-intelligence/api-client/pull/229#discussion_r440377797>
> :
>
> >          """
>          return lib.get_geo_centre(self.access_token, self.api_host, region_id)
>
> -    def get_geojsons(self, region_id, descendant_level=None):
> +    def get_geojsons(self, region_id, descendant_level=None, zoom_level=7):
>
> just to confirm, we used to return simplified shapefile for all requests.
> is it intended to change the default to detailed boundaries table?
> @nemozen <https://github.com/nemozen>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gro-intelligence/api-client/pull/229#pullrequestreview-430906303>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADTN4VGRPC645WRMQFELHFLRWZURXANCNFSM4NZSZ3LQ>
> .
>

nemozen committed:
make args required in lib

They are optional in GroClient which sets defaults before calling lib.
Having defaults at both levels is error prone, so  make them required in the lower level lib.

nemozen committed:
pass zoom_levevel to lib

nemozen committed:
mock_get_geojsons

nemozen committed:
Merge branch 'CLEWS-27183' of https://github.com/gro-intelligence/api-client into CLEWS-27183

nemozen committed:
no default for lib function zoom_level

nemozen committed:
in get_geo_centre, pass params via dict instead of url string

cn1036 commented:
> Yes, we want to keep the current server-side default, so we don't change client side performance of the web app. for api clients, we want to default to more detailed boundaries (as with shp files), and give the client the option to reduce the zoom level if higher level of accuracy is not needed.
> […](#)
> On Mon, Jun 15, 2020 at 2:51 PM cn1036 ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In api/client/gro_client.py <[#229 (comment)](https://github.com/gro-intelligence/api-client/pull/229#discussion_r440377797)> : > """ return lib.get_geo_centre(self.access_token, self.api_host, region_id) - def get_geojsons(self, region_id, descendant_level=None): + def get_geojsons(self, region_id, descendant_level=None, zoom_level=7): just to confirm, we used to return simplified shapefile for all requests. is it intended to change the default to detailed boundaries table? @nemozen <https://github.com/nemozen> — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#229 (review)](https://github.com/gro-intelligence/api-client/pull/229#pullrequestreview-430906303)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADTN4VGRPC645WRMQFELHFLRWZURXANCNFSM4NZSZ3LQ> .

itsjohnward commented:
^ I see the shippable run just failed on "pluggy.manager.PluginValidationError: Plugin 'pytest_cov' could not be loaded"

That error should be fixed by https://github.com/gro-intelligence/api-client/pull/232. Merging from dev to re-run tests.

itsjohnward committed:
Merge branch 'development' into CLEWS-27183

cn1036 committed:
update unit tests

itsjohnward committed:
Merge branch 'development' into CLEWS-27183

nemozen committed:
Merge branch 'development' into CLEWS-27183

nemozen committed:
Merge branch 'CLEWS-27183' of https://github.com/gro-intelligence/api-client into CLEWS-27183

itsjohnward commented on api/client/lib.py:
```diff
@@ -538,26 +538,26 @@ def lookup_belongs(access_token, api_host, entity_type, entity_id):
-def get_geojsons(access_token, api_host, region_id, descendant_level=None):
-    url = '/'.join(['https:', '', api_host, 'v2/geocentres?includeGeojson=True&regionIds={}'.format(
-        region_id)])
+def get_geojsons(access_token, api_host, region_id, descendant_level, zoom_level):
```
These lib functions, `get_geo_centre`, `get_geojsons`, and `get_geojson` are not covered by any tests, and since this PR is adding code to them, it has decreased test coverage a bit. Add these to `lib_test.py`?

cn1036 replied:
yeah you're right. working on it now
